### PR TITLE
Revert "Set up key backup using non-deprecated APIs (#12005)"

### PR DESCRIPTION
**Pull request by BillCarsonFr**
This reverts commit df11b90fd658167003410d8df4c60fb54a059c91, and also adds a regression test.

Fixes https://github.com/element-hq/element-web/issues/26814 

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Revert "Set up key backup using non-deprecated APIs (#12005)" ([\#12102](https://github.com/matrix-org/matrix-react-sdk/pull/12102)). Fixes element-hq/element-web#26814.

### CODE REVIEW

File: src/SecurityManager.ts [hunk 1]
```diff
@@ -341,22 +341,9 @@ export async function withSecretStorageKeyCache<T>(func: () => Promise<T>): Prom
...
  * @param {bool} [forceReset] Reset secret storage even if it's already set up
- * @param {bool} [setupNewKeyBackup] Reset secret storage even if it's already set up
  */
-export async function accessSecretStorage(
-    func = async (): Promise<void> => {},
-    forceReset = false,
-    setupNewKeyBackup = true,
-): Promise<void> {
-    await withSecretStorageKeyCache(() => doAccessSecretStorage(func, forceReset, setupNewKeyBackup));
-}
```
Comment (richvdh) [inline src/SecurityManager.ts:352]: This change was introduced in https://github.com/matrix-org/matrix-react-sdk/pull/12059. I'd suggest leaving it in place (without `setupNewKeyBackup`, of course).

**Review by richvdh — CHANGES_REQUESTED** [review]

**Comment by richvdh** [comment]
As I understand it, this will re-open https://github.com/element-hq/element-web/issues/26323.

File: src/SecurityManager.ts (hunk 1 above)
Comment (BillCarsonFr) [reply src/SecurityManager.ts]: Done https://github.com/matrix-org/matrix-react-sdk/pull/12102/commits/35806b5bc793b233011ceeccbf8cd09fba6745a5

**Review by richvdh — APPROVED** [review]
lgtm

File: playwright/e2e/crypto/utils.ts [hunk 2]
```diff
@@ -103,12 +103,29 @@ export async function checkDeviceIsCrossSigned(app: ElementAppPage): Promise<voi
...
+
+    // expand the advanced section to see the active version in the reports
+    const summary = page.locator(".mx_SecureBackupPanel_advanced");
+    await summary.locator("..").click();
+
+    const cacheDecryptionKeyStatusElement = page.locator(".mx_SecureBackupPanel_statusList tr:nth-child(2) td");
+    await expect(cacheDecryptionKeyStatusElement).toHaveText("cached locally, well formed");
+
+    const serverVersion = await page.locator(".mx_SecureBackupPanel_statusList tr:nth-child(5) td").textContent();
+    expect(serverVersion.trim()).toBe(expectedBackupVersion + " (Algorithm: m.megolm_backup.v1.curve25519-aes-sha2)");
```
Comment (richvdh) [inline playwright/e2e/crypto/utils.ts:125]:
```suggestion
    await expect(page.locator(".mx_SecureBackupPanel_statusList tr:nth-child(5) td")).toHaveText(expectedBackupVersion + " (Algorithm: m.megolm_backup.v1.curve25519-aes-sha2)");
```

File: playwright/e2e/crypto/utils.ts [hunk 3]
```diff
@@ -103,12 +103,29 @@ export async function checkDeviceIsCrossSigned(app: ElementAppPage): Promise<voi
...
+
+    const activeVersion = await page.locator(".mx_SecureBackupPanel_statusList tr:nth-child(6) td").textContent();
+    expect(activeVersion.trim()).toBe(expectedBackupVersion);
```
Comment (richvdh) [inline playwright/e2e/crypto/utils.ts:128]:
```suggestion
    await expect(page.locator(".mx_SecureBackupPanel_statusList tr:nth-child(6) td")).toHaveContent(expectedBackupVersion);
```

File: playwright/e2e/crypto/utils.ts [hunk 4]
```diff
@@ -103,12 +103,29 @@ export async function checkDeviceIsCrossSigned(app: ElementAppPage): Promise<voi
...
  */
-export async function checkDeviceIsConnectedKeyBackup(page: Page) {
+export async function checkDeviceIsConnectedKeyBackup(page: Page, expectedBackupVersion: string): Promise<void> {
     await page.getByRole("button", { name: "User menu" }).click();
     await page.locator(".mx_UserMenu_contextMenu").getByRole("menuitem", { name: "Security & Privacy" }).click();
     await expect(page.locator(".mx_Dialog").getByRole("button", { name: "Restore from Backup" })).toBeVisible();
+
+    // expand the advanced section to see the active version in the reports
+    const summary = page.locator(".mx_SecureBackupPanel_advanced");
+    await summary.locator("..").click();
```
Comment (richvdh) [inline playwright/e2e/crypto/utils.ts:119]: combine these?

```suggestion
    await page.locator(".mx_SecureBackupPanel_advanced").locator("..").click();
```

File: playwright/e2e/crypto/verification.spec.ts [hunk 5]
```diff
@@ -32,6 +32,7 @@ import { Bot } from "../../pages/bot";
 
 test.describe("Device verification", () => {
     let aliceBotClient: Bot;
+    let expectedBackupVersion: string;
```
Comment (richvdh) [inline playwright/e2e/crypto/verification.spec.ts:35]:
```suggestion

    /** The backup version that was set up by the bot client. */
    let expectedBackupVersion: string;
```

**Review by richvdh — CHANGES_REQUESTED** [review]

**Review by richvdh — APPROVED** [review]
lgtm.

@t3chguy are you happy to bypass the coverage gate on this?

File: playwright/pages/ElementAppPage.ts [hunk 6]
```diff
@@ -51,19 +51,6 @@ export class ElementAppPage {
         return this.settings.closeDialog();
     }
 
-    public async getClipboard(): Promise<string> {
-        return await this.page.evaluate(() => navigator.clipboard.readText());
-    }
```
Comment (t3chguy) [inline playwright/pages/ElementAppPage.ts:56]: Why is this going away?

Comment (t3chguy) [reply playwright/pages/ElementAppPage.ts:56]: Oh we also have `getClipboardText` - fun

**Review by t3chguy — APPROVED** [review]

**Pull request merged**